deepinv/models/bilateral.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import annotations
+import torch
+from torch import Tensor
+from .base import Denoiser
+
+
+class BilateralFilter(Denoiser):
+    r"""
+    Bilateral filter.
+
+    The bilateral filter as it was introduced in :footcite:t:`tomasi1998bilateral`
+    where each output pixel is a normalized weighted average of neighboring pixels in a spatial window.
+    The weights factor into a spatial kernel and a range kernel:
+
+        \hat{x}_i = (1 / W_i) * \sum_{j \in \omega(i)} k_d(i - j) k_r(x_i - x_j) x_j
+
+    with Gaussian spatial kernel k_d and Gaussian range kernel k_r.
+    The spatial standard deviation $\sigma_d$ controls how fast the kernel
+    decays with distance, while the range standard deviation $\sigma_r$
+    controls how strongly intensity differences are penalized.
+
+    |sep|
+
+    >>> import deepinv as dinv
+    >>> x = dinv.utils.load_example("butterfly.png")
+    >>> physics = dinv.physics.GaussianNoise()
+    >>> y = physics(x)
+    >>> model = dinv.models.BilateralFilter()
+    >>> x_hat = model(y,sigma_d=1.1,sigma_r=0.3,window_size=9)
+    >>> dinv.metric.PSNR()(x_hat, x) > 26.13
+    tensor([True])
+    """
+
+    def forward(
+        self,
+        x: Tensor,
```

Comment:
```suggestion
        x: Tensor,
        sigma: torch.Tensor | float | None = None,
        *,
```

The denoiser interface expects a 2nd positional parameter for the noise level when it's known. Adding it will help prevent bugs when the denoiser is combined with other parts of the library

The * is to make parameters following it keyword-only - we probably don't want to rely on parameter positions here in case we add new ones in the future